pyecoregen/templates/module.py.tpl

```diff
@@ -20,12 +20,49 @@ eClassifiers = {}
 getEClassifier = partial(Ecore.getEClassifier, searchspace=eClassifiers)
 
 {%- for c in element.eClassifiers if c is type(ecore.EEnum) -%}
+{% if loop.first and textX %}
+class EEnum(Ecore.EEnum):
+
+    def __init__(self, name=None, default_value=None, literals=None, **kwargs):
+        super().__init__(name, default_value, literals)
+        self.__dict__.update(kwargs)
+
+    @property
+    def value(self):
+        ''' returns the value of the attribute '''
+        for attr in [a for a in self.__dict__ if
+                     not a.startswith('__') and not
+                     a.startswith('_tx_')]:
+            obj = getattr(self, attr)
+            if obj and isinstance(obj, str):
+                return obj
+{% endif %}
 {{ modutil.generate_enum(c) }}
 {%- endfor %}
 {% for c in element.eClassifiers if c is type(ecore.EDataType) -%}
+{% if loop.first and textX %}
+class EDataType(Ecore.EDataType):
+
+    def __new__(cls, *args, **kwargs):
+        self = Ecore.EDataType.__new__(cls, *args, **kwargs)
+        self.__name__ = args[0]
+        return self
+
+{% endif %}
 {{ modutil.generate_edatatype(c) }}
 {%- endfor %}
 
 {%- for c in classes -%}
+{% if loop.first and textX %}
```

Comment:
If I understood correctly (and I remember from textX), the idea is to put all unknown `kwargs` as simple instance attribute?
Anyway, I would propose:

```python
def __init__(self, **kwargs):
    super().__init__(**kwargs)
    for key, value in kwargs.items():
        if key not in self.__dict__:
            continue
        setattr(self, key, value)
```
What do you think?

Reply:
Yap, textX puts all unknown kwargs as instance attributes.

I change it, like the readability of your code.